### PR TITLE
Replace mapval with go-lookslike

### PR DESCRIPTION
**Pull request by andrewvc**
This cleans up some tech debt since the mapval package has been renamed
and moved to https://github.com/elastic/go-lookslike , we should stop
using the now older/outdated mapval package.

### CODE REVIEW

File: CHANGELOG-developer.next.asciidoc [hunk 1]
```diff
@@ -25,6 +25,7 @@ The list below covers the major changes between 7.0.0-rc2 and master only.
 - Build docker and kubernetes features only on supported platforms. {pull}13509[13509]
 - Need to register new processors to be used in the JS processor in their `init` functions. {pull}13509[13509]
 - The custom beat generator now uses mage instead of python, `mage GenerateCustomBeat` can be used to create a new beat, and `mage vendorUpdate` to update the vendored libbeat in a custom beat. {pull}13610[13610]
+- Altered all remaining uses of mapval to use the renamed and enhanced version: [go-lookslike](https://github.com/elastic/go-lookslike) instead, which is a separate project. The mapval tree is now gone. {pull}14165[14165]
```
Comment (andrewkroh) [inline CHANGELOG-developer.next.asciidoc]: Just checking if this is asciidoc syntax? _Lookslike_ markdown.

Comment (andrewvc) [reply CHANGELOG-developer.next.asciidoc]: Ah, good catch, will fix!

**Comment by andrewvc** [comment]
jenkins, retest this please

**Comment by andrewvc** [comment]
jenkins, test this please

**Pull request merged**